ocaml/squeezed/lib/squeeze.ml

```diff
@@ -252,14 +246,17 @@ module Stuckness_monitor = struct
         then
           state.stuck <- true
       )
-      host.domains ;
+      domains ;
     (* Clear out dead domains just in case someone keeps *)
     (* one of these things around for a long time.       *)
-    let live_domids = List.map (fun domain -> domain.domid) host.domains in
+    let module IntSet = Set.Make (Int) in
+    let live_domids =
+      DomainSet.fold (fun d acc -> IntSet.add d.domid acc) domains IntSet.empty
```

Comment:
Just a question: would the `Set.filter_map` work better in this case?

Reply:
The type of `DomainSet.filter_map` is `(elt -> elt option) -> DomainSet.t -> DomainSet.t`, it will return `DomainSet`.
In this case, we want `IntSet` finally.

Reply:
In practice, they are equivalent because their ordering function is equivalent